ambry-store/src/main/java/com.github.ambry.store/StorageManager.java

```diff
@@ -153,10 +153,28 @@ public Store getStore(PartitionId id) {
    * @return the {@link DiskManager} corresponding to the given {@link PartitionId}, or {@code null} if no DiskManager was found for
    *         that partition
    */
-  DiskManager getDiskManager(PartitionId id) {
+  public DiskManager getDiskManager(PartitionId id) {
     return partitionToDiskManager.get(id);
   }
 
+  /**
+   * Check if a certain disk is unavailable.
+   * @param diskManager the {@link DiskManager} to check.
+   * @return {@code true} if the disk is unavailable. {@code false} if not.
+   */
+  public boolean isDiskUnavailable(DiskManager diskManager) {
```

Comment:
the input for this should be a `Disk` not a `DiskManager`. This function serves no purpose otherwise. Someone who has the `diskManager` can just call `areAllStoresDown()`

Reply:
true, will make the change.